fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/values/MessageHelpers.java

```diff
@@ -360,7 +360,9 @@ public static <M extends Message> Object transformMessage(@Nonnull final FDBReco
             } else {
                 if (currentMessage != null) {
                     final var currentFieldType = Verify.verifyNotNull(currentRecordType.getField(messageFieldDescriptor.getIndex())).getFieldType();
-                    final var fieldResult = NullableArrayTypeUtils.unwrapIfArray(getFieldOnMessage(currentMessage, messageFieldDescriptor), currentFieldType);
+                    var fieldResult = messageFieldDescriptor.isRepeated() || !messageFieldDescriptor.getType().equals(Descriptors.FieldDescriptor.Type.MESSAGE) ?
```

Comment:
Can you add a comment? What is this doing?

Reply:
With the addition of UUID, now there are certain messages that are represented differently in the `runtime`, essentially all the ones defines in the 'tuple_fields.proto'. For those cases, we do not want to get the `runtime` value but the actual message. 

Hence, if the field is a message, we ask for the raw message, else, we get the field value.